Extension/tslint.json

```diff
@@ -1,31 +1,184 @@
+        "no-internal-module": true,
+        "no-invalid-template-strings": false,
+        "no-invalid-this": true,
+        "no-irregular-whitespace": true,
+        "no-magic-numbers": false,
+        "no-mergeable-namespace": true,
+        "no-misused-new": true,
+        "no-namespace": true,
+        "no-non-null-assertion": true,
+        "no-null-keyword": false,
+        "no-object-literal-type-assertion": false,
+        "no-parameter-properties": false,
+        "no-parameter-reassignment": false,
+        "no-redundant-jsdoc": true,
+        "no-reference": true,
+        "no-reference-import": true,
+        "no-require-imports": false,
+        "no-return-await": true,
+        "no-shadowed-variable": false,
+        "no-sparse-arrays": true,
+        "no-string-literal": false,
+        "no-string-throw": false,
+        "no-submodule-imports": false,
+        "no-switch-case-fall-through": true,
+        "no-this-assignment": true,
+        "no-trailing-whitespace": true,
+        "no-unbound-method": false,
+        "no-unnecessary-callback-wrapper": true,
+        "no-unnecessary-class": false,
+        "no-unnecessary-initializer": true,
+        "no-unnecessary-qualifier": true,
+        "no-unnecessary-type-assertion": false,
+        "no-unsafe-any": false,
+        "no-unsafe-finally": true,
+        "no-unused-expression": true,
         "no-unused-variable": true,
+        "no-use-before-declare": false,
         "no-var-keyword": true,
-        "one-line": [true, "check-catch", "check-finally", "check-else", "check-open-brace", "check-whitespace"],
+        "no-var-requires": true,
+        "no-void-expression": false,
+        "number-literal-format": true,
+        "object-literal-key-quotes": false,
+        "object-literal-shorthand": false,
+        "object-literal-sort-keys": false,
+        "one-line": [
+            true,
+            "check-catch",
+            "check-finally",
+            "check-else",
+            "check-open-brace",
+            "check-whitespace"
+        ],
+        "one-variable-per-declaration": true,
+        "only-arrow-functions": false,
+        "ordered-imports": false,
+        "prefer-conditional-expression": false,
+        "prefer-const": false,
+        "prefer-for-of": false,
+        "prefer-function-over-method": false,
+        "prefer-method-signature": true,
+        "prefer-object-spread": true,
+        "prefer-readonly": false,
+        "prefer-switch": false,
+        "prefer-template": false,
+        "prefer-while": true,
+        "promise-function-async": false,
         "promise-must-complete": true,
+        "quotemark": false,
+        "radix": false,
+        "restrict-plus-operands": false,
+        "return-undefined": false,
         "semicolon": true,
+        "space-before-function-paren": false,
+        "space-within-parens": true,
+        "strict-boolean-expressions": false,
+        "strict-type-predicates": true,
+        "switch-default": false,
```

Comment:
Having a default case is good to catch errors. 

https://palantir.github.io/tslint/rules/switch-default/